modules/node/src/auth/auth.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { arrayify, HDNode, hexlify, randomBytes, verifyMessage } from "ethers/utils";
+
+import { ChannelRepository } from "../channel/channel.repository";
+import { RedisProviderId } from "../constants";
+import { CLogger, isValidHex, isXpub } from "../util";
+
+const logger = new CLogger("AuthService");
+const nonceLen = 16;
+const nonceTTL = 2 * 60 * 60 * 1000;
+
+const badToken = (warning: string): any => {
+  logger.warn(warning);
+  return { err: `Invalid token` } as any;
+};
+
+const badSubject = (warning: string): any => {
+  logger.warn(warning);
+  return { err: `Invalid subject` } as any;
+};
+
+@Injectable()
+export class AuthService {
```

Comment:
This is probably a good place to add the JWT vending.

The JWT needs to include the following claims:
```
{
  "iat":1570112805,
  "exp":1601648805,
  "aud":"nats.connext.network",
  "sub":"someXpub",
  "permissions":[

  ]
}
```

Insert NATS `permissions` for publish, subscribe and request/response. Then sign it.

Reply:
`aud` isn't really needed above.